core/src/main/java/org/apache/carbondata/core/indexstore/Blocklet.java

```diff
@@ -92,7 +95,13 @@ public String getFilePath() {
         blocklet.blockletId == null;
   }
 
-  @Override public int hashCode() {
+  @Override
+  public String toString() {
+    return String.format("[shard:%s, blockletId:%s]", filePath, blockletId);
```

Comment:
You can use CarbonTablePath.getShardName to get the shard name

Reply:
The shard name in `toString()` is exactly the one in `Blocklet` itself

Reply:
updated